### ISSUE TITLE
Fix S2589 FN: Capture numerical constrain knowledge

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#7780]
### Description

SE does not capture the knowledge about the numerical constraints of variables.

### Repro steps

```csharp
a = 0 switch
{
    1 => true            // FN, because there's no knowledge that a != 1
    , _ => false
};
```

### Expected behavior

Should raise S2589 issue.
